service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/process/DockerManagedBox.scala

```diff
@@ -80,6 +85,82 @@ class DockerManagedBox(
 
 
     //=========================================================================
+
+
+    override def registerServiceNode(serviceNodeName: String, componentsDistributorUrl: String): Either[String, BoxInformation] = {
+        distributorClients.put(serviceNodeName, JsonRestClient.create[ComponentsDistributor](componentsDistributorUrl))
+        Right(boxInformation)
+    }
+
+
+    override def registerCustomNodeComponentType(serviceNodeName: String, componentName: String): Either[String, String] = {
+        val componentNameFile = new File(s"$InnerPath/distributives/$componentName").getAbsoluteFile
+        if (componentNameFile.exists()) {
+            logger.info(s"Component $componentName is already exists on a box manager")
+            Right("Success")
+        }
+        else {
+            logger.info(s"Component $componentName isn't on a box manager. Querying distributor client...")
+            for {
+                distributorClient <- distributorClients.get(serviceNodeName).toRight(s"Service node $serviceNodeName is not registered in box manager")
+                distributiveBase64 <- distributorClient.getComponentTypeDistributive(componentName)
+                distributive <- Try(Base64.getDecoder.decode(distributiveBase64)).toEither.left.map(_.getMessage)
+                _ = Util.untarFile(distributive, s"$InnerPath/distributives/")
```

Comment:
use `componentNameFile` here instead i.e. do NOT expect that tarball will contain inner folder with name matching `componentName` also this could be security breach as this way any component could override content of some other

Reply:
At this point I made check only whether **componentType** folder exists or not. Not checking the content of the folder.